buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -773,9 +773,28 @@ class BuildPlugin implements Plugin<Project> {
     }
 
     static void applyCommonTestConfig(Project project) {
+        String defaultParallel = 'auto'
+        if (project.file("/proc/cpuinfo").exists()) {
+            Map<String, Integer> socketToCore = [:]
+            String currentID = ""
+            project.file("/proc/cpuinfo").readLines().forEach({ line ->
+                if (line.contains(":")) {
+                    List<String> parts = line.split(":", 2).collect({it.trim()})
+                    String name = parts[0], value = parts[1]
+                    if (name == "physical id") {
+                        currentID = value
+                    }
+                    if (name == "cpu cores") {
+                        assert currentID.isEmpty() == false
+                        socketToCore[currentID] = Integer.valueOf(value)
+                    }
+                }
+            })
+            defaultParallel = socketToCore.values().sum().toString();
```

Comment:
This gets the number of logical cores, I think. I thought we wanted the number of physical cores here. So `socketToCore.size()`. I think.

Reply:
Socket is for when you might have multiple CPU socket, as in multiple CPU chips, thus the sum.